enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,320 @@
+Destinations are either on or off cluster endpoints such as the cluster logging provided Elasticsearch, an organization's Kafka message bus, or other log aggregation solutions (e.g. Splunk, Graylog).
+
+## Motivation
+
+Organizations desire to reuse their existing enterprise log solutions to store container logs.  Providing a declarative mechanism by which administrators define a log destination simplifies their operational burden.  They are able to take advantage of log collection infrastructure with minimal configuration changes. The cluster logging stack is able to deploy a collector to each OKD node that is configured with the necessary permissions to collect the logs, add container metadata (e.g. labels) and ship them to the specified endpoint.
+
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that identifies log sources and their desired outputs.
+* Send logs to an Elasticsearch cluster not managed by the cluster logging infrastructure
+* Send logs to a fluentd instance not managed by the cluster logging infrastructure via fluentd's secureforward protocol
+* Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)
+* Support TLS between the collector and destination if so configured
+* Use the same log forwarding logic to write logs to a specified output or to the default storage managed by cluster logging (e.g. Elasticsearch).
+
+We will be successful when:
+
+* an administrator is able to deploy their own log aggregation service
+* specifies this service in log forwarding configuration
+* the service receives logs from a OKD cluster logging deployed collection agent  
+
+### Non-Goals
+
+Log forwarding intends to be opinionated about the source of log messages; it will explicitly define what constitutes "application logs" or "infrastructure logs". It will not allow configuration of additional sources without further design considerations and how those sources may impact the collecting agent. Log forwarding is not intended to provide a complex routing solution as one might achieve by using a custom collector configuration or a messaging solution (e.g. kafka).
+
+## Proposal
+
+Log forwarding will provide a declarative way to specify the outputs for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more outputs. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
```

Comment:
this does not include the audit log right?  so "logs.audit" is also a future source (i believe that's distinct from container engine audit events)

Reply:
Right.  I don't know if/where we should talk about the possibility of future pipeline sources.